tests/drivers/flash/src/main.c

```diff
@@ -93,7 +93,9 @@ static void *flash_driver_setup(void)
 	if (!is_buf_clear) {
 		/* erase page */
 		rc = flash_erase(flash_dev, page_info.start_offset,
-				 page_info.size);
+				 MAX(page_info.size, EXPECTED_SIZE));
```

Comment:
This test was made with silent assumption that the flash page sizes >= EXPECTED_SIZE.

value used (based on `MAX(page_info.size, EXPECTED_SIZE)`) should be additionally aligned so it contains whole flash pages.

Reply:
@nvlsianpu can you please clarify : do you mean aligning EXPECTED_SIZE with page_info.size ?

Reply:
aligned to the size of all pages to which EXPECTED_SIZE belongs, for instance:
```
{--------------------EXPECTED_SZIE----------------------}
{----page0----}{--page1---}{----------------page2---------------}
```
used erase size should be sum of these 3 pages.

Reply:
Ok, 
with this sample, the page_size is fixed so the size to erase could simply be` int(EXPECTED_SIZE / page_info.size + 1)`

Reply:
@nvlsianpu Can you take a look?

Reply:
my remarks were not applied correctly. Additional not needed flash page might be erased, also this assume all touched pages have same size.

I know that despite that this fix will work on every device - this is just test. Deficiencies can be fixed in the future if needed.